core/src/HIP/Kokkos_HIP_Instance.cpp

```diff
@@ -388,6 +390,39 @@ void HIPInternal::finalize() {
     m_team_scratch_current_size = 0;
     m_team_scratch_ptr          = nullptr;
   }
+  if (nullptr != d_driverWorkArray) {
+    HIP_SAFE_CALL(hipHostFree(d_driverWorkArray));
+    d_driverWorkArray = nullptr;
+  }
+}
+
+char *HIPInternal::get_next_driver(size_t driverTypeSize) const {
+  if (d_driverWorkArray == nullptr) {
+    HIP_SAFE_CALL(
+        hipHostMalloc(&d_driverWorkArray,
+                      m_maxDriverCycles * m_maxDriverTypeSize * sizeof(char),
+                      hipHostMallocNonCoherent));
+  }
+  if (driverTypeSize > m_maxDriverTypeSize) {
+    // fence handles the cycle id reset for us
+    fence();
+    HIP_SAFE_CALL(hipHostFree(d_driverWorkArray));
+    m_maxDriverTypeSize = driverTypeSize;
+    if (m_maxDriverTypeSize % 128 != 0)
```

Comment:
Where does `128` come from?

Reply:
Hmm -- Leopold wrote that bit, I assume it was some attempt at padding / alignment, but I'll check w/ him.  It's probably safe to remove since it's pretty arbitrary.